### PR TITLE
Set allowPrivilegeEscalation to false for the service orchestrator

### LINKED ISSUE #4409: Set allowPrivilegeEscalation=false in executor
To be able to run Seldon in more restricted environments we need the executor container to have the `securityContext.allowPrivilegeEscalation` set to `false`.

Currently no value is set and according to the Kubernetes documentation the default value is not clearly specified. As a consequence some policies are blocking model deployments(like the [privilege escalation one in Azure AKS](https://learn.microsoft.com/en-us/azure/aks/policy-reference)).

As the container does not need privilege escalation it could have the property set [here](https://github.com/pdstrnadJC/seldon-core/blob/master/operator/controllers/seldondeployment_engine.go#L428) or allow the user to configure it via `svcOrch`.

<sup>Winfried Umbrath winfried.umbrath@mercedes-benz.com, Mercedes-Benz Tech Innovation GmbH, [imprint](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)</sup>

### PR DESCRIPTION
**Pull request by wumbrath**
**What this PR does / why we need it**:
The executor/service orchestrator container does not need privilege escalation. As a security best practice and to allow for
runnning seldon in more restricted environments this PR sets the securityContext.allowPrivilegeEscalation property to false.

**Which issue(s) this PR fixes**:

Fixes #4409

**Special notes for your reviewer**:

<sup>The program was tested solely for our own use cases, which might differ from yours.</sup>
<sup>Winfried Umbrath winfried.umbrath@mercedes-benz.com, Mercedes-Benz Tech Innovation GmbH, [imprint](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)</sup>

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @wumbrath. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **majolo**
You can assign the PR to them by writing `/assign @majolo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by wumbrath** [comment]
Tests are working for me locally, not sure why it fails on github

**Comment by seldondev** [comment]
@wumbrath: Cannot trigger testing until a trusted user reviews the PR and leaves an `/ok-to-test` message.

<details>

In response to [this](https://github.com/SeldonIO/seldon-core/pull/4427#issuecomment-1303484274):

>/retest


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by cliveseldon** [comment]
Looks good - will run tests

**Comment by seldondev** [comment]
@wumbrath: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 8378e6b0c40d360da2da791f1dab6787618e07ca | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4427-integration-6gsk6) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by cliveseldon** [comment]
Just 1 failure - will try again

Screenshot_2022-11-08_10-40-57

Screenshot_2022-11-08_10-40-40

**Comment by seldondev** [comment]
@wumbrath: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 8378e6b0c40d360da2da791f1dab6787618e07ca | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4427-integration-bddnq) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by cliveseldon** [comment]
Seemed to pass integration but fail on shutdown:
Screenshot_2022-11-08_14-32-49

**Comment by seldondev** [comment]
@wumbrath: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 8378e6b0c40d360da2da791f1dab6787618e07ca | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4427-integration-rbxtt) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by axsaucedo** [comment]
It seems the previous failure was a rolling upgrade test and now it's a different one so we can confirm it was a flaky test so merging now 👍 
/approve

**Review by axsaucedo — APPROVED** [review]

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/4427#pullrequestreview-1173710576 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/4427#pullrequestreview-1173710576 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
